segmentation/utils/spatial_analysis_test.py

```diff
@@ -49,13 +45,9 @@ def make_distance_matrix(enrichment_type):
     elif enrichment_type == "negative":
         # This creates a distance matrix where there are two groups of cells significant for 2 different
         # markers that are not located near each other (not within the dist_lim).
-        dist_mat_neg = np.zeros((60, 60))
-        dist_mat_neg[20:40, :20] = 300
-        dist_mat_neg[:20, 20:40] = 300
-        dist_mat_neg[40:50, :40] = 50
-        dist_mat_neg[:40, 40:50] = 50
-        dist_mat_neg[50:60, :50] = 200
-        dist_mat_neg[:50, 50:60] = 200
+
+        dist_mat_neg = synthetic_spatial_datagen.direct_init_dist_matrix(num_A=20, num_B=20, num_C=20,
```

Comment:
Why does this pass our tests? Doesn't this make it so that all cells are equally spaced away from one another? This shouldn't result in any negative enrichment

Reply:
I must be misunderstanding what negative enrichment is. I made the assumption that it meant that the cells of type 1 and type 2 are on average farther away from each other than dist_lim according to the documentation under the `enrichment_type == 'negative'` case in make_distance_matrix. So, I didn't think that the insignificant marker cells (which I'll call type 3) mattered. However, I'm guessing more is involved.

The tests for negative enrichment, in a nutshell, seem dependent on the fact that there are on average an equal number or fewer number of cells within the dist_lim of close_num as compared to close_num_rand. In theory, and if my understanding is correct, this means that if the negative enrichment test matrices are initialized such that the mean distance between type 1 and type 2 cells is about the same as type 1 and type 3 cells, we should be seeing the desired results.

A quick fix could be changing distr_random in the negative matrix initialization to have a lower mean than that of distr_AB, but as mentioned, it's quite possible more is involved that I missed.

Reply:
That's close, but not quite right. The dist_lim determines whether a cell is close to another cell. In positive enrichment, we are looking for cell types A and B to have more cells that are close to one another than would be expected by chance. 

For negative enrichment, we are looking for cell types A and B to have fewer cells close to one another than would be expected by chance. 

So in Jay's original hardcoded data, cell types A and B only had 1 or two cells that were close to one other out of 20, but the overall population prevelance of cells that were close to one another was around 20%. Hence, they were negatively enriched. 

You put that for negative enrichment, the tests check that 'there are on average an equal number or fewer number of cells within the dist_lim of close_num as compared to close_num_rand.' If this is the case, we should change the tests, as what this is supposed to be testing is strict inferiority, not less than or equal to.

Does that all make sense?

Reply:
Ok, I see. I'll change the test to in spatial_analysis_utils to reflect that. Also will change the distribution parameters in the negatively-enriched matrix to ensure cell types A and B have fewer cells close to one another than expected.